Add download files section (#24)
AlexsLemonade/scpca-docs: docs/download_files.md

Every download will also include a `libraries_metadata.csv` file containing metadata associated with each library included in the download.

## Unfiltered Counts Matrix 

The unfiltered counts matrix, `<scpca_library_id>_unfiltered.rds`, is delivered as a RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html).
Within the `SingleCellExperiment` is the counts matrix, where the rows correspond to genes or features and the columns correspond to cell barcodes. 
Here, all potential cell barcodes that are identified after running [Alevin-fry](processing_information.html/#alignment-and-quantification-using-alevin-fry) are included in the counts matrix. 
Summary statistics for each cell and gene can be found in the `colData` and `rowData` of the `SingleCellExperiment` object.
The `SingleCellExperiment` object also contains metadata about that particular library, including the versions of Salmon and Alevin-fry used for pre-processing, information about the index used for transcriptome alignment, and parameters used for Alevin-fry. 
See also [Using the provided RDS files in R.](link to FAQ)

> ## Filtered Counts Matrix

Author: I like the idea of putting these under one heading of `Gene expression data`. In terms of describing the colData and rowData, I wasn't sure how much detail we wanted to include here and am thinking that increasing the level of detail might belong in a section where we discuss addition of QC and metadata. I think #14 starts to address this, but we might want to add information about adding per cell and per feature QC in the processing section, rather than here?

|-----------------|----------------------------------------------------------------|
| scpca_sample_id | Sample ID in the form `SCPCS000000`                            |
| scpca_library_id | Library ID in the form `SCPCL000000`                          |
| submitter_id    | Original sample identifier from submitter                      |
| participant_id  | Original participant id, if there are multiple samples from the same participant                                                                          |
| submitter       | Submitter name/id                                              |
| age             | Age at time sample was obtained                                |
| sex             | Sex of patient sample is obtained from                      |
| diagnosis       | Tumor type                                                     |
| subdiagnosis    | Subcategory of diagnosis or mutation status (if applicable)   |
| tissue_location | Where tumor sample was located                                 |
| disease_timing  | What stage of disease was the sample obtained? At diagnosis or recurrence? |

> Additional metadata may also be included that will be specific to the disease type. Examples of this include treatment or outcome. 

Reviewer: I think we want to be clear that this is as much project-specific as disease type. Not sure I love the wording.
```suggestion
Additional metadata may also be included, specific to the disease type and experimental design of the project. Examples of this include treatment or outcome. 
```

> For each library, there will also be a `<scpca_library_id>_metadata.json`, containing processing associated metadata.

Reviewer: ```suggestion
For each library, is also a `<scpca_library_id>_metadata.json` file that contains processing-associated metadata.
```

# Downloadable files

> The ScPCA portal download packages include gene expression data, a QC report, and associated metadata for each processed sample.

Reviewer: ```suggestion
The ScPCA Portal download packages include gene expression data, a QC report, and associated metadata for each processed sample.
```

These files are delivered as a zip file.

> The folder structure within the zip file is determined by selecting to download all samples associated with a project or selecting individual samples.  

Reviewer: Trying to reword this to flow a bit better.
```suggestion
The folder structure within the zip file is determined by whether individual samples or all samples associated with a project are selected for download.  
```

## Download folder structure for project downloads: 
_add in illustration of file structure for project_ 

## Download folder structure for individual sample downloads: 
_add in illustration of file structure for sample downloads_ 

Sample folders (indicated by the `SCPCS` prefix) contain the files for all libraries (`SCPCL` prefix) derived from that biological sample. 
Most samples only have one library that has been sequenced. 

> See {ref}`the difference between samples and libraries <faq:what is the difference between a sample and a library?>` for more information.

Reviewer: I felt like the link text here was a bit confusing.
```suggestion
See the {ref}`FAQ section about samples and libraries <faq:what is the difference between a sample and a library?>` for more information.
```

> The files associated with each library are:

Reviewer: Not sure about the wording here, but trying to be a bit more explicit here.
```suggestion
The files associated with each library are (shown for a library with ID `SCPCL000000`):
```

- An unfiltered counts file: `SCPCL000000_unfiltered.rds`, 
- A filtered counts file: `SCPCL000000_filtered.rds`, 
- A quality control report: `SCPCL000000_qc.html`, 

> - A metadata file: `SCPCL000000_metadata.json`. 

Reviewer: Just to be sure the markdown is parsed correctly.
```suggestion
- A metadata file: `SCPCL000000_metadata.json`. 

```

> Every download also includes a `libraries_metadata.csv` file containing metadata associated with each library included in the download.

Reviewer: Trying to clarify here that this is for multiple libraries, not one per library. 

```suggestion
Every download also includes a single `libraries_metadata.csv` file containing metadata for all libraries included in the download.
```

## Gene expression data

> Gene expression data is available in two forms - as an unfiltered counts file or filtered counts file.

Reviewer: ```suggestion
Gene expression data is provided in two forms - as an unfiltered counts file and a filtered counts file.
```

> The unfiltered counts file, `SCPCL000000_unfiltered.rds`, is delivered as a RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html).

Reviewer: Simplifying a bit.
```suggestion
The unfiltered counts file, `SCPCL000000_unfiltered.rds`, is an RDS file containing a [`SingleCellExperiment` object](http://bioconductor.org/books/3.13/OSCA.intro/the-singlecellexperiment-class.html).
```

Within the `SingleCellExperiment` object is the counts matrix, where the rows correspond to genes or features and the columns correspond to cell barcodes. 
Here, all potential cell barcodes that are identified after running {ref}`Alevin-fry <processing_information:alignment and quantification using alevin-fry>` are included in the counts matrix. 

> Summary statistics for each cell and gene can be found in the `colData` and `rowData` slots of the `SingleCellExperiment` object, respectively.
> The `SingleCellExperiment` object also contains metadata about that particular library, including the versions of Salmon and Alevin-fry used for pre-processing, information about the index used for transcriptome alignment, and parameters used for Alevin-fry. 

Reviewer: I am not sure how much detail we want to put into this, since we will cover it in a separate section. I would definitely cut down the second sentence, as I am suggesting here. I'm also removing the references to `colData` and `rowData`, but I feel less strongly about that. 

```suggestion
The object also includes summary statistics for each cell barcode and gene, as well as metadata about that particular library, such as the reference index and software versions used for mapping and quantification. 
```

> See also {ref}`Using the provided RDS files in R <faq:how do i use the provided RDS files in r?>`.

Reviewer: I think I would move this sentence to after the "see gene expression file contents", or combine the two.

> The filtered counts file is also delivered as a RDS file containing a `SingleCellExperiment` object.
> Following filtering using {ref}`emptyDrops <processing_information:filtering cells>`, the filtered counts matrix and updated summary statistics for each cell and gene are output to the `SCPCL000000_filtered.rds` file.
> As a result, this file only contains cell barcodes that are considered true cells.

Reviewer: I'm trying to keep this paragraph more parallel with the previous one.
```suggestion
The filtered counts file, `SCPCL000000_filtered.rds` is also an RDS file containing a `SingleCellExperiment` object with the same structure as above.
The cells in this file are those that remain after filtering using {ref}`emptyDrops <processing_information:filtering cells>`.
As a result, this file only contains cell barcodes that are likely to correspond to true cells.
```

> See {ref}`gene expression file contents <gene_expression_file_contents:gene expression file contents>` for more information.

Reviewer: Adding a bit more detail.
```suggestion
See {ref}`gene expression file contents <gene_expression_file_contents:gene expression file contents>` for more information about the contents of the `SingleCellExperiment` objects and the included statistics and metadata.
```

## QC Report 

The included QC report serves as a general overview of each library, including processing information, summary statistics and general visualizations of cell metrics.

## Metadata

The `libraries_metadata.csv` file is a comma-separated table with one row per library and the following columns. 

| column_id       | contents                                                       |
|-----------------|----------------------------------------------------------------|
| scpca_sample_id | Sample ID in the form `SCPCS000000`                            |
| scpca_library_id | Library ID in the form `SCPCL000000`                          |
| seq_unit        | `cell` for single-cell samples or `nucleus` for single-nucleus samples |
| technology      | 10X kit used to process library                                |
| filtered_cell_count | Number of cells after filtering with `emptyDrops`          |
| submitter_id    | Original sample identifier from submitter                      |

> | participant_id  | Original participant id, if there are multiple samples from the same participant                                                                          |

Reviewer 2: I don't think I've seen this one before. Where is it coming from?

| submitter       | Submitter name/id                                              |
| age             | Age at time sample was obtained                                |

> | sex             | Sex of patient sample is obtained from                      |

Reviewer: Consistent tense, mostly.
```suggestion
| sex             | Sex of patient that the sample was obtained from                      |
```

| diagnosis       | Tumor type                                                     |
| subdiagnosis    | Subcategory of diagnosis or mutation status (if applicable)   |

> | tissue_location | Where tumor sample was located                                 |

Reviewer: Maybe? I don't think this is really ambigous, but it read strangely to me. 
```suggestion
| tissue_location | Where in the body the tumor sample was located                                 |
```

| disease_timing  | What stage of disease was the sample obtained? At diagnosis or recurrence? |

> Additional metadata may also be included, specific to the disease type and experimental design of the project. Examples of this include treatment or outcome. 

Reviewer: Just breaking at sentence.
```suggestion
Additional metadata may also be included, specific to the disease type and experimental design of the project.
Examples of this include treatment or outcome. 
```
